yjit/src/core.rs

```diff
@@ -2397,6 +2407,81 @@ fn new_pending_branch(jit: &mut JITState, gen_fn: BranchGenFn) -> PendingBranchR
     branch
 }
 
+c_callable! {
+    /// Generated code calls this function with the SysV calling convention.
+    /// See [gen_leave_compile].
+    fn leave_compile_hit(ec: EcPtr) -> *const u8 {
+        with_vm_lock(src_loc!(), || {
+            // Return compiled code, or code to return Qundef
+            leave_compile_hit_body(ec).unwrap_or(
+                CodegenGlobals::get_stub_exit_code().raw_ptr()
+            )
+        })
+    }
+}
+
+/// This is called after a frame is popped. The current CFP points to the caller,
+/// and this function attempts to compile and execute the caller.
+fn leave_compile_hit_body(ec: EcPtr) -> Option<*const u8> {
```

Comment:
Worth documenting in the comment that this is used for exceptions and for compiling somewhere in the middle of the caller?

Reply:
:+1: https://github.com/ruby/ruby/pull/8171/commits/b55383c85d31f2df9f5dbbd5775d70dc1b8eb3f2